### ISSUE TITLE
Remove `componentWillReceiveProps()` in `Addon`

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#6770]
See also: #6188

---

The `Addon` component is by far our largest component, so let's remove the `componentWillReceiveProps()` method separately.

**For QA:** only the add-on detail pages might be affected, please make sure there is no regression.

**Comment by ioanarusiczki** [comment]
@willdurand I did not see regressions on the add-on detail page.

I've noticed a behavior that is reproducible on AMO prod too - the "Report this add-on for abuse" form closes when the "Mozilla add-on policy" link is clicked and then I return back on AMO. If I navigate to "Read all reviews" page and return on the detail page the abuse form is still displayed. This is also reproducible for "Report this user for abuse". 
Reporting for abuse forms will close when links that don't open other AMO pages are clicked (I tested with those from the footer of the page besides the "Mozilla add-on policy" link).

Let me know if I should file an issue for this.

![report this add-on for abuse form](https://user-images.githubusercontent.com/33448286/47990616-a03e2400-e0f0-11e8-8444-4ec9c3a22a07.gif)
